src/transport/smtp/async_transport.rs

```diff
@@ -17,7 +27,20 @@ use crate::Tokio1Transport;
 
 #[allow(missing_debug_implementations)]
 pub struct AsyncSmtpTransport<C> {
-    // TODO: pool
+    #[cfg(all(
+        feature = "tokio1-pool",
+        not(any(feature = "tokio02", feature = "async-std1"))
+    ))]
+    marker_: PhantomData<C>,
+    #[cfg(all(
+        feature = "tokio1-pool",
+        not(any(feature = "tokio02", feature = "async-std1"))
+    ))]
+    inner: Pool<AsyncSmtpClient<Tokio1Connector>>,
```

Comment:
This should take `C`, since in theory lettre could be built with multiple async runtimes turned on, and the actual runtime for any instance of `AsyncSmtpTransport` would depend on `C`. tokio1-pool would then mean that for the pool to work this `AsyncSmtpTransport` would have to be in a tokio 1 context

Reply:
Done. I had to restrict the trait bounds for `AsyncSmtpConnector` since `ManageConnection` had specific requirements.